routers/api/v1/api.go

```diff
@@ -266,287 +266,295 @@ func mustAllowPulls(ctx *context.Context) {
+		m.Group("/members", func() {
+			m.Get("", org.ListMembers)
+			m.Combo("/:username").Get(org.IsMember).
+				Delete(reqToken(), reqOrgOwnership(), org.DeleteMember)
+		})
+		m.Group("/public_members", func() {
+			m.Get("", org.ListPublicMembers)
+			m.Combo("/:username").Get(org.IsPublicMember).
+				Put(reqToken(), reqOrgMembership(), org.PublicizeMember).
+				Delete(reqToken(), reqOrgMembership(), org.ConcealMember)
+		})
+		m.Combo("/teams", reqToken(), reqOrgMembership()).Get(org.ListTeams).
+			Post(bind(api.CreateTeamOption{}), org.CreateTeam)
+		m.Group("/hooks", func() {
+			m.Combo("").Get(org.ListHooks).
+				Post(bind(api.CreateHookOption{}), org.CreateHook)
+			m.Combo("/:id").Get(org.GetHook).
+				Patch(reqOrgOwnership(), bind(api.EditHookOption{}), org.EditHook).
+				Delete(reqOrgOwnership(), org.DeleteHook)
+		}, reqToken(), reqOrgMembership())
+	}, orgAssignment(true))
+	m.Group("/teams/:teamid", func() {
+		m.Combo("").Get(org.GetTeam).
+			Patch(reqOrgOwnership(), bind(api.EditTeamOption{}), org.EditTeam).
+			Delete(reqOrgOwnership(), org.DeleteTeam)
+		m.Group("/members", func() {
+			m.Get("", org.GetTeamMembers)
+			m.Combo("/:username").
+				Put(reqOrgOwnership(), org.AddTeamMember).
+				Delete(reqOrgOwnership(), org.RemoveTeamMember)
 		})
+		m.Group("/repos", func() {
+			m.Get("", org.GetTeamRepos)
+			m.Combo("/:orgname/:reponame").
+				Put(org.AddTeamRepository).
+				Delete(org.RemoveTeamRepository)
+		})
+	}, orgAssignment(false, true), reqToken(), reqOrgMembership())
 
-		m.Group("/admin", func() {
-			m.Group("/users", func() {
-				m.Post("", bind(api.CreateUserOption{}), admin.CreateUser)
-				m.Group("/:username", func() {
-					m.Combo("").Patch(bind(api.EditUserOption{}), admin.EditUser).
-						Delete(admin.DeleteUser)
-					m.Post("/keys", bind(api.CreateKeyOption{}), admin.CreatePublicKey)
-					m.Post("/orgs", bind(api.CreateOrgOption{}), admin.CreateOrg)
-					m.Post("/repos", bind(api.CreateRepoOption{}), admin.CreateRepo)
-				})
+	m.Any("/*", func(ctx *context.Context) {
+		ctx.Error(404)
+	})
+
+	m.Group("/admin", func() {
+		m.Group("/users", func() {
+			m.Post("", bind(api.CreateUserOption{}), admin.CreateUser)
+			m.Group("/:username", func() {
+				m.Combo("").Patch(bind(api.EditUserOption{}), admin.EditUser).
+					Delete(admin.DeleteUser)
+				m.Post("/keys", bind(api.CreateKeyOption{}), admin.CreatePublicKey)
+				m.Post("/orgs", bind(api.CreateOrgOption{}), admin.CreateOrg)
+				m.Post("/repos", bind(api.CreateRepoOption{}), admin.CreateRepo)
 			})
-		}, reqAdmin())
+		})
+	}, reqAdmin())
+}
+
+// RegisterRoutes registers all v1 APIs routes to web application.
+// FIXME: custom form error response
+func RegisterRoutes(m *macaron.Macaron) {
+	m.Get("/swagger", misc.Swagger) //Render V1 by default
+
+	m.Group("/v1", func() {
+		apiRoutes(m)
+	}, context.APIContexter())
+
+	m.Group("/v3", func() {
+		apiRoutes(m)
```

Comment:
Some of these routes are not GitHub-compatible. They shouldn't be added to `v3/` 😕